### ISSUE TITLE
Rule option proposal: less-than support for explicit-length-check

### ISSUE THREAD

**Issue by Garbee** [sindresorhus/eslint-plugin-unicorn#2009]
### Description

Supporting `less-than` as an option for how to style the length check would allow for this rule to be used in conjunction with ones like `etc/prefer-less-than` [1] without conflicting. Right now, if the style for checks prefers less than and the zero check is not set to do strict checking, the two errors collide and one has to be ignored over the other.

Less than is purely a style choice, just like greater than or strict equation.

[1] https://github.com/cartant/eslint-plugin-etc/blob/main/docs/rules/prefer-less-than.md

### Fail

```js
function foo() {
  var data = [];
  while (data.length > 0) {
	console.log('Something for the block');
  }
}
```

### Pass

```js
function foo() {
  var data = [];
  while (0 < data.length) {
	console.log('Something for the block');
  }
}
```

**Comment by notramo** [comment]
https://twitter.com/BenLesh/status/1397593619096166400

It's indeed helpful to use only one of these operators through the entire source code. In western (left to right) writing systems, the less-than operator is easier to understand, as e.g. a sorted array also has smaller values on the left and incrementing as we read it forward.

**Comment by fregante** [comment]
While I don't like it, I think this is a bug. 

The rule requires an "explicit length check" and `0 < x.length` is exactly that, but it's marked as invalid. Then [eslint/yoda](https://eslint.org/docs/latest/rules/yoda) would take care of flipping that for those who want it.

**Comment by sindresorhus** [comment]
I agree it's a bug, but I also don't think it's worth spending the effort fixing this or the increased complexity in the logic. Such a style is simply too niche. We can document that the rule does not work well with yoda style.
